README.md

```diff
@@ -1,37 +1,107 @@
-- [OPTIONAL] Boost.Test. Used when `BUILD_TESTING` is enabled.
+- Linux with GCC 11
+- Linux with Clang 14
+- MacOS 11 with Apple-clang 13.0.0
+- Windows with Visual Studio 16
 
-# Building
+## Requirements
 
-STLab is a standard CMake project. See the [running CMake](https://cmake.org/runningcmake) tutorial for an introduction to this tool.
+- A standards-compliant C++14, C++17, or C++20 compiler
+- **Use with a compiler in C++14-compliant mode** requires Boost.Optional and Boost.Variant >= 1.74.0
+- **Building** the library requires CMake 3.23 or later
+- **Testing or developing** the library requires Boost.Test >= 1.74.0
+
+## Building
+
+STLab is a standard CMake project. See the [running CMake](https://cmake.org/runningcmake) tutorial
+for an introduction to this tool.
+
+### Preparation
+
+1. Create a build directory outside this library's source tree.  In this guide, we'll use a sibling
+   directory called `BUILD`.
+
+2. If you are using the library in C++14-compliant mode or need to run the test suite, be sure you
+   have the necessary parts of Boost >= 1.74.0 installed.  Linux distributions usually make a
+   suitable version available through standard package managers.  On macOS or Linux, you can easilly
+   install Boost using [Homebrew](https://brew.sh/).  To install Boost on Windows, you can use
+   Boost's [binary installers](https://sourceforge.net/projects/boost/files/boost-binaries/).
+
+3. Install a version of CMake >= 3.23.  If you are on Debian or Ubuntu Linux you may need to use
+   `snap` to find one that's new enough.
+
+4. If you are using MSVC, you may need to set environment variables appropriately for your target
+   architecture by invoking `VCVARSALL.BAT` with an appropriate option.
+
+### Configure
+
+Run CMake in the root directory of this project, setting `../BUILD` as your build directory.  The
+basis of your command will be
+
+```
+cmake -S . -B ../BUILD -DCMAKE_BUILD_TYPE=# SEE BELOW
+```
+
+but there are other options you may need to append in order to be successful.  Among them:
+
+* `-DCMAKE_BUILD_TYPE=`[`Release`|`Debug`] to build the given configuration (required).
+* `-DCMAKE_CXX_STANDARD=`[`14`|`17`|`20`|`23`] to build with compliance to the given C++ standard.
+* `-DSTLAB_USE_BOOST_CPP17_SHIMS=ON` if you are building in C++14 compliant mode.
+* `-DBUILD_TESTING=OFF` if you only intend to build, but not test, this library.
+* `-DSTLAB_NO_STD_COROUTINES=TRUE` if your compiler doesn't have standard coroutine support. This applies to Apple platforms and when CMAKE_CXX_STANDARD is set to 14 or 17.```
+* `-DBoost_USE_STATIC_LIBS=TRUE` if you will be testing on Windows.
+
+We also suggest the installation of [Ninja](https://ninja-build.org/) and its use by adding
+`-GNinja` to your cmake command line… but ninja is not required.
+
+A typical invocation might look like this:
+
+```
+cmake -S . -B ../BUILD -GNinja -DCMAKE_CXX_STANDARD=17 -DBUILD_TESTING=OFF
```

Comment:
This is missing the `-DCMAKE_BUILD_TYPE=Release`.

Reply:
Good catch.  
```suggestion
cmake -S . -B ../BUILD -GNinja -DCMAKE_CXX_STANDARD=17 -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF
```